pkg/infrastructure/azure/azure.go

```diff
@@ -26,3 +47,312 @@ func InitializeProvider() clusterapi.Provider {
 func (p *InfraProvider) Name() string {
 	return "azure"
 }
+
+// PreProvision is called before provisioning using CAPI controllers has begun.
+func (p *InfraProvider) PreProvision(ctx context.Context, in clusterapi.PreProvisionInput) error {
```

Comment:
When implementing one of these interfaces, add  a check like `var _ clusterapi.PreProvider = (*Provider)(nil)` so we will get a build-time error if this Provider doesn't implement the interface it thinks it's implementing

Reply:
Fixed.